### ISSUE TITLE
Separate the save to the cloud from the map parsing

### ISSUE THREAD

**Issue by SharonHadar** [TechnionYearlyProject/BTW#72]
this separation will make it possible to postpone the saving untill the user will ask for it. or even do it in parallel to the map building and using. the user experience will improve for the lack of need to wait.

**Comment by SharonHadar** [comment]
this was implemented with a new class in the DB module called GeoJasonToJavaParser
the way to use this class from the GenerateCityController is to call the method parseMap(String) (the string is the map in geojson) of the BTWDataBase object. the parsing is done and then the flow continues as before. the saving will be done by calling to the method saveMap(String) (the string is the map in geojson)of the BTWDataBase object.
